### PR TITLE
Navigate back press fix

### PR DESCRIPTION
**Pull request by jojonarte**
Fix for Issue #334

### CODE REVIEW

**Comment by codeguru42** [comment]
@jojonarte What is the state of this branch? Do you need to write any tests or are the existing tests sufficient?

**Comment by jojonarte** [comment]
@codeguru42 I want to know your opinion with my tests on this pr. As of the moment I believe I've test as much as I know for backpress.

**Comment by codeguru42** [comment]
@jojonarte Please edit the message for your first commit to something more descriptive. It wasn't immediately obvious to me that it has the tests for this issue. I will take a look at them in more detail later this week.

File: android/src/main/java/bbct/android/common/activity/BaseballCardList.java
Comment (codeguru42) [inline android/src/main/java/bbct/android/common/activity/BaseballCardList.java]: This commit appears to duplicate changes from the #430. Is this intentional? Which branch do these changes belong on? They should only be on one or the other. Please amend one of the branches to fix this.

File: android/src/main/java/bbct/android/common/activity/MainActivity.java
Comment (codeguru42) [inline android/src/main/java/bbct/android/common/activity/MainActivity.java]: The same goes for all the changes in this file.

**Review by codeguru42 — CHANGES_REQUESTED** [review]

File: android/src/main/java/bbct/android/common/activity/MainActivity.java
Comment (jojonarte) [reply android/src/main/java/bbct/android/common/activity/MainActivity.java]: I assumed we would go back to the BaseballCardList when we pressback from BaseballCardDetails. Isn't that right?

File: android/src/main/java/bbct/android/common/activity/BaseballCardList.java
Comment (jojonarte) [reply android/src/main/java/bbct/android/common/activity/BaseballCardList.java]: done :)

File: android/src/main/java/bbct/android/common/activity/MainActivity.java
Comment (codeguru42) [reply android/src/main/java/bbct/android/common/activity/MainActivity.java]: Yes, that is correct. But since this exact change already exists in PR #428, it should not be repeated here. Instead, it should be merged in from that branch. I will make time to finish reviewing the tests in #428 this week and hopefully get it merged into `android/devel`.

**Review by codeguru42 — CHANGES_REQUESTED** [review]

File: android/src/main/java/bbct/android/common/activity/MainActivity.java
Comment (jojonarte) [reply android/src/main/java/bbct/android/common/activity/MainActivity.java]: Okay, then I can merge with that then update this PR right?

Comment (codeguru42) [reply android/src/main/java/bbct/android/common/activity/MainActivity.java]: Let's put this PR on hold for now until I merge #428. After that, the best solution will probably be a rebase.

**Comment by codeguru42** [comment]
When you have time to continue, this should be the next thing. Merge with `devel/android` and resolve conflicts. Then let me know when it is ready for review.

**Comment by jojonarte** [comment]
@codeguru42  okay thanks

**Comment by codeguru42** [comment]
@jojonarte Is there anything else you need to do on this? Or is it ready for a final review?

**Comment by jojonarte** [comment]
@codeguru42 there's still some things I have to fix.

**Comment by codeguru42** [comment]
@jojonarte Okay. Just ping me when you are finished.

**Comment by jojonarte** [comment]
@codeguru42 it's ready for review now.

**Comment by codeguru42** [comment]
Does 9c82268 address the problem from #429?

**Comment by jojonarte** [comment]
@codeguru42 it doesn't yet, I'll fix that also don't review yet. Will let you know.

**Comment by codeguru42** [comment]
@jojonarte Just to be clear, do you mean you will fix that on this branch? If so, let's close #429.

**Comment by jojonarte** [comment]
@codeguru42  yes

**Comment by jojonarte** [comment]
@codeguru42  okay

**Comment by jojonarte** [comment]
Is there anything left for this PR?

**Comment by codeguru42** [comment]
@jojonarte Thanks for following up on this. This PR is good for the moment. I am currently making some changes and updates that may require additional changes. I'll let you know when that happens. In the mean time, feel free to work on #308 or another issue.

**Comment by codeguru42** [comment]
This is obsolete after the Kotlin and Jetpack compose rewrite.

**Pull request closed without merging**
